rotkehlchen/chain/ethereum/modules/aave/aave.py

```diff
@@ -163,6 +176,190 @@ def _get_balances(
+            elif event.event_subtype in (HistoryEventSubType.PAYBACK_DEBT, HistoryEventSubType.REMOVE_ASSET, HistoryEventSubType.REWARD):  # noqa: E501
+                if event.extra_data is not None and 'is_liquidation' in event.extra_data:
+                    total_earned[event.asset] += event.balance
+                if event.event_subtype == HistoryEventSubType.REWARD:
+                    atokens_balances[event.asset] += event.balance
+                else:
+                    historical_borrow_balances[event.asset] += event.balance.amount
+            elif event.event_subtype == HistoryEventSubType.LIQUIDATE:
+                # At liquidation you lose the collateral asset
+                total_lost[event.asset] += event.balance
+            elif event.event_subtype == (HistoryEventSubType.GENERATE_DEBT, HistoryEventSubType.RECEIVE_WRAPPED):  # noqa: E501
+                atokens_balances[event.asset] += event.balance
+            elif event.event_subtype == (HistoryEventSubType.PAYBACK_DEBT, HistoryEventSubType.RETURN_WRAPPED):  # noqa: E501
+                atokens_balances[event.asset] -= event.balance
+
+            if event.asset in atokens_balances:
+                amount_diff = atokens_balances[event.asset].amount - prev_balance.amount
+                usd_price = query_usd_price_zero_if_error(
+                    asset=event.asset,
+                    time=ts_ms_to_sec(event.timestamp),
+                    location=f'aave interest event {event.event_identifier} from history',
+                    msg_aggregator=self.msg_aggregator,
+                )
+                earned_atoken_balances[event.asset] += Balance(amount=amount_diff, usd_value=amount_diff * usd_price)  # noqa: E501
+
+        for b_asset, amount in historical_borrow_balances.items():
+            borrow_balance = balances.borrowing.get(cast(CryptoAsset, b_asset), None)
+            this_amount = amount
+            if borrow_balance is not None:
+                this_amount += borrow_balance.balance.amount
+
+            usd_price = Inquirer().find_usd_price(b_asset)
+            total_lost[b_asset] = Balance(
+                # add total_lost amount in case of liquidations
+                amount=total_lost[b_asset].amount + this_amount,
+                usd_value=this_amount * usd_price,
+            )
+
+        # calculate the earned interest
+        self._calculate_interest_and_profit(
+            user_address=user_address,
+            balances=balances,
+            total_earned_atokens=earned_atoken_balances,
+        )
+        return total_lost, total_earned, earned_atoken_balances
+
+    def _calculate_interest_and_profit(
+            self,
+            user_address: ChecksumEvmAddress,
+            balances: AaveBalances,
+            total_earned_atokens: dict[Asset, Balance],
+    ) -> None:
+        """
+        Calcualte total earned from aave events. `total_earned_atokens` is modified inside
+        this function
+        """
```

Comment:
This function comes from the previous implementation. The ONLY change made was to store the abi in a variable to avoid loading it always from DB